### ISSUE TITLE
Blur/pixelation every 2 seconds

### ISSUE THREAD

**Issue by TwitchTVBlaku** [Xaymar/OBS-AMD-Advanced-Media-Framework#300]
I'm pretty sure this is something that is common so I don't think you'll need my settings to understand. But i'm just curious what causes the blur every ~2 seconds. I see this on just about every Twitch stream that uses AMD, and NVENC as well. It also happens on QSync. Even with a still image, you can often notice this slight blur/pixelation. It is an instant blur that quickly reverts the image to normal, but it happens every 2-3 seconds with 100% consistency. It's not a huge issue but during intensive scenes all this pixelation starts to add up. 

Hopefully you can give me some insight, as this seems to be an issue with QS/NVENC/AMD while x264 is fine. I think you'll know what I'm talking about anyways.

Cheers

**Comment by Glitchvid** [comment]
It's probably due to being bandwidth limited and the encoder deciding to put more bits into the I-frames than the P-frames.  
As you get "further" from each I-frame while you're bit-rate starved, the image is going to get more "blocky" or "muddy".
For the AMD HEVC encoder, I think the rate at which this is set is the "Keyframe Interval".

There are a few options, but a lot of this is just the limitation of AVC/HEVC and the encoders ability to actually work at low bit-rates. I'd try upping the keyframe interval, make sure the video is CBR, and make sure the encoder profile is set to "Quality".

Just my 2¢, I'm not an expert or contributor to this project.

**Comment by Xaymar** [comment]
The blockyness (pixelation, blur) you are describing is an effect from a bitrate that is too low for the encoder or configuration used. The strength of this effect depends on the encoder used, NVENC (on Pascal and newer) and x264 suffer the least, while QSV and AMF are really bad at properly recovering from these situations.

From what I know, AMF works based on the "leaky bucket" bitrate control model. It works, but looks visually awful since it only looks at bitrate instead of the visual output. Also it is really awful at recovering unless there are scenes that allow it to recover (like low color variety and low motion). The only one who could fix this is AMD, and the Pre-Pass & VBAQ attempt have not increased visual quality much (difference of 0.1%).

So, in short: AMF, QSV and NVENC (Kepler and earlier) are useful for recording or high bitrates. NVENC (Pascal) and x264 are useful for streaming and low bitrates.

**Comment by Xaymar** [comment]
Closing as more than a week has passed without a reply.
